### PR TITLE
feat(Dropdown): add ControlledDropdown component

### PR DESCRIPTION
**Pull request by ajkl2533**
To provide support for more usecases we introduce the ControlledDropdown
component. This component allows user to take complete control of the Dropdown.

### CODE REVIEW

File: src/components/Dropdown/ControlledDropdown.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  { value: 'Leo', label: 'Leo' },
+  { value: 'Jack', label: 'Jack' },
+];
+
+export const UsingWithAnotherComponentControlled: Story<DropdownProps> = (
+  args,
+) => {
+  const [currentValue, setCurrentValue] = useState(null);
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [triggerEl, setTriggerEl] = useState<HTMLElement>(null);
```
Comment (andrasbudayssc) [inline src/components/Dropdown/ControlledDropdown.stories.tsx:47]: Would useRef work here?

File: src/components/Dropdown/Dropdown.stories.tsx [hunk 2]
```diff
@@ -236,9 +241,9 @@ export const UsingWithAnotherComponent: Story<DropdownProps> = () => (
     <CreatableSelect
       createNewLabel="Buy new kitten"
       options={options}
-      defaultIsMenuOpen
       isMenuPositionRelative
       isMulti
+      menuIsOpen
```
Comment (andrasbudayssc) [inline src/components/Dropdown/Dropdown.stories.tsx:246]: I don't see the corresponding change to this prop. May have missed it but worth checking it out.

**Review by andrasbudayssc — COMMENTED** [review]
LGTM

**Review by andrasbudayssc — APPROVED** [review]

**Comment by andrasbudayssc** [comment]
@ajkl2533 Good to go once tests are fixed 👍

File: src/components/Dropdown/Dropdown.stories.tsx (hunk 2 above)
Comment (ajkl2533) [reply src/components/Dropdown/Dropdown.stories.tsx:246]: This is property from the `react-select`, it's supported, but it's not documented (thus use without any warranty)

File: src/components/Dropdown/ControlledDropdown.stories.tsx (hunk 1 above)
Comment (ajkl2533) [reply src/components/Dropdown/ControlledDropdown.stories.tsx:47]: nope, only `node`

**Pull request merged**